core/chain_makers.go

```diff
@@ -238,58 +243,79 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
 	return blocks, receipts
 }
 
-func makeHeader(chain consensus.ChainReader, parent *types.Block, state *state.StateDB, engine consensus.Engine) *types.Header {
-	var time uint64
-	if parent.Time() == 0 {
-		time = 10
-	} else {
-		time = parent.Time() + 10 // block time is fixed at 10 seconds
+func makeHeader(config *params.ChainConfig, chain consensus.ChainReader, parent *types.Block, order int, state *state.StateDB, engine consensus.Engine) *types.Header {
+	// return genesis block as itself for first block in chain
+	if parent.Header().Number[0].Cmp(big.NewInt(0)) == 0 {
+		return parent.Header()
```

Comment:
I'm not sure these lines do what the comment indicates. If the parent is genesis (aka this is block 1), then return genesis instead of a new header. Isn't this just going to always short circuit to return the genesis header?